pkg/cmd/psa/parser.go

```diff
@@ -0,0 +1,65 @@
+package psa
+
+import (
+	"regexp"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const headerString = "existing pods in namespace"
+
+var (
+	// wordsInParenthesis parses words from parenthesis.
+	wordsInParenthesis = regexp.MustCompile(`"([^"]+)"`)
+	// submatch in contrast to the whole match.
+	submatch = 1
+	// firstMatch is the first match.
+	firstMatch = 0
+	// secondMatch is the second match.
+	secondMatch = 1
+)
+
+// parseWarnings parses the warnings that are returned by the API request and
+// creates a PodSecurityViolation out of it, that contains the namespace, level,
+// pod name and violations.
+// We expect to have
+//
+// Example Warning Slice:
+// [0] existing pods in namespace "p0t-sekurity-namespace" violate the new PodSecurity enforce level "restricted:latest"
+// [1] p0t-sekurity-pod: allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+// OR
+// [1] p0t-sekurity-pod (and 1 other pod): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+// OR
+// [1] p0t-sekurity-pod (and 10 other pods): allowPrivilegeEscalation != false, unrestricted capabilities, runAsNonRoot != true, seccompProfile
+func parseWarnings(warnings []string) *PodSecurityViolation {
+	if len(warnings) == 0 {
+		return nil
+	}
+
+	var psv PodSecurityViolation
+
+	// There should be exactly 2 warnings, but lets not rely on the order.
```

Comment:
why does the number of warnings matter?  why not scan all warnings?  What if we add a new warning?

Reply:
I did it on all, until I realized there are only two, so I wanted to make it explicit. But lets change it to being more dynamic again and add a comment.

Reply:
I would like it to make it explicit. I was surprised by the fact that it is exactly two. The follow up code works for a header and non-header warnings.

Should I remove the warning?

Reply:
Removing it and updating the bellow comment.